prow/gerrit/adapter/adapter.go

```diff
@@ -491,16 +504,37 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		}
 	case client.New:
 		var presubmits []config.Presubmit
+		// Gerrit server might be unavailable intermittently, retry inrepoconfig
+		// processing for increased reliability.
 		for attempt := 0; attempt < inRepoConfigRetries; attempt++ {
 			presubmits, err = c.inRepoConfigCacheHandler.GetPresubmits(trimmedHostPath, func() (string, error) { return baseSHA, nil }, func() (string, error) { return change.CurrentRevision, nil })
 			if err == nil {
 				break
 			}
 		}
-		if err := c.handleInRepoConfigError(err, instance, change); err != nil {
+		if err != nil {
+			// Reports error back to Gerrit. handleInRepoConfigError is
+			// responsible for not sending the same message again and again on
+			// the same commit.
+			if postErr := c.handleInRepoConfigError(err, instance, change); postErr != nil {
+				logger.WithError(postErr).Error("Failed reporting inrepoconfig processing error back to Gerrit.")
+			}
+			// There is no need to keep going when failed to get inrepoconfig
+			// jobs.
+			// Imagining the scenario that:
+			// - Commit #abc triggered static job job-A, inrepoconfig jobs job-B
+			// and job-C
+			// - Both job-B and job-C failed
```

Comment:
This part of the scenario is not relevant/necessary. The jobs don't need to have previously run, this can happen on the first commit.